src/plugins/footnotes/index.ts

```diff
@@ -285,15 +296,25 @@ export default (props: PluginProps) => {
       decorations: (state) => {
         const decorations: Decoration[] = []
 
-        const isInTableElementFooter = hasParentNodeOfType(
+        const tableElementFooter = findParentNodeOfType(
```

Comment:
This change is causing the issue of inserting new regular footnotes and adding new table footnotes as well.

Reply:
Can you please elaborate more?

Reply:
In some manner, it's impacting the returned position. I've encountered this issue previously and discovered that employing a similar approach affects the selection of the position/node. I'm uncertain why this might occur, but directly passing the node/parent within the "state.doc.descendants((node, pos, parent) => {" function proved to be the solution. I suggest attempting to revert some of the older code changes and pass the target node differently, rather than relying on the findParentNodeOfType method.

Reply:
@nghazzawi-atypon  Do you mean that finding required node by iterating doc.descendants should be used instead of findParentNodeOfType?

Reply:
@mbartenev-atypon exactly !
please take a look at the the implementation of deleting general/table footnotes on master branch

Reply:
`findParentNodeOfType` has no problem. It does correctly and effectively find the node and its position.

The issue lies specifically with the widget position in a footnote node compared to a general table note.

For footnotes, the target node is the footnote itself, so the widget position is expected to be there + 1.

![Screenshot 2024-05-09 at 2 56 12 PM](https://github.com/Atypon-OpenSource/manuscripts-body-editor/assets/33928809/aa44c6ad-501d-4db2-a41e-02e01c9e0e2f)

In the case of the general note, the target node is the tableElementFooter node, so the widget position is expected to be there + 2.

![Screenshot 2024-05-09 at 2 54 23 PM](https://github.com/Atypon-OpenSource/manuscripts-body-editor/assets/33928809/7e13755f-7c55-4b87-97cf-eddae04223b5)

Please review the latest commit. Also, please note that once we have a special node type for the general table note that would simplify things.